tmc-plugin/src/fi/helsinki/cs/tmc/actions/DownloadAdaptiveExerciseAction.java

```diff
@@ -0,0 +1,72 @@
+package fi.helsinki.cs.tmc.actions;
+
+import fi.helsinki.cs.tmc.core.TmcCore;
+import fi.helsinki.cs.tmc.core.domain.Exercise;
+import fi.helsinki.cs.tmc.core.domain.ProgressObserver;
+import fi.helsinki.cs.tmc.core.events.TmcEventBus;
+import fi.helsinki.cs.tmc.coreimpl.BridgingProgressObserver;
+import fi.helsinki.cs.tmc.model.ProjectMediator;
+import fi.helsinki.cs.tmc.model.TmcProjectInfo;
+import fi.helsinki.cs.tmc.ui.ConvenientDialogDisplayer;
+import fi.helsinki.cs.tmc.utilities.BgTask;
+import fi.helsinki.cs.tmc.utilities.BgTaskListener;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.concurrent.Callable;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionRegistration;
+import org.openide.util.NbBundle.Messages;
+
+@ActionID(category = "TMC", id = "fi.helsinki.cs.tmc.actions.DownloadAdaptiveExerciseAction")
+@ActionRegistration(displayName = "#CTL_DownloadAdaptiveExerciseAction")
+@ActionReference(path = "Menu/TM&C", position = -200)
+@Messages("CTL_DownloadAdaptiveExerciseAction=DownloadAdaptiveExerciseAction")
+public final class DownloadAdaptiveExerciseAction implements ActionListener {
+
+    private static final Logger logger = Logger.getLogger(DownloadAdaptiveExerciseAction.class.getName());
+    
+    private final ProjectMediator projectMediator;
+    private final ConvenientDialogDisplayer dialogs;
+    private final TmcEventBus eventBus;
+    
+    public DownloadAdaptiveExerciseAction() {
+        this.projectMediator = ProjectMediator.getInstance();
+        this.dialogs = ConvenientDialogDisplayer.getDefault();
+        this.eventBus = TmcEventBus.getDefault();
+    }
+    
+    @Override
+    public void actionPerformed(ActionEvent e) {
+
+        logger.log(Level.WARNING, "Init adaptive exercise downloading");
```

Comment:
I'm less familiar with the logging standards in TMC-Netbeans, but my intuition says that this should probably be level "info" rather than "warning". Synchronize with the rest of the project.